pkg/instrumentation/golang.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instrumentation
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+)
+
+const (
+	kernelDebugVolume = "kernel-debug"
+)
+
+func injectGolangSDK(_ logr.Logger, golangSpec v1alpha1.Golang, pod corev1.Pod) corev1.Pod {
+	// skip instrumentation if share process namespaces is explicitly disabled
+	if pod.Spec.ShareProcessNamespace != nil && *pod.Spec.ShareProcessNamespace == false {
+		return pod
+	}
+	execPath, ok := pod.Annotations[annotationGolangExecPath]
```

Comment:
could we get this information from the registered [v1.Container command](https://github.com/kubernetes/api/blob/v0.24.3/core/v1/types.go#L2301-L2310) itself? That might limit the number of containers to `1`, but should be fine for now. What do you think?

Reply:
the command in the container spec can be empty if I am not mistaken - e.g. reply on default from the container image. 

Even inspecting the container image CMD or entrypoint is not sufficient. Many containers run scripts before executing the final process.